src/Microsoft.Tye.Core/ProjectReader.cs

```diff
@@ -30,94 +39,136 @@ public static async Task ReadProjectDetailsAsync(OutputContext output, FileInfo
                 throw new ArgumentNullException(nameof(project));
             }
 
+            EnsureMSBuildRegistered(projectFile);
+
             using (var step = output.BeginStep("Reading Project Details..."))
             {
-                await EvaluateMSBuildAsync(output, projectFile, project);
+                EvaluateProject(output, projectFile, project);
 
                 if (!SemVersion.TryParse(project.Version, out var version))
                 {
-                    output.WriteInfoLine($"No version or invalid version 'application.Version' found, using default.");
+                    output.WriteInfoLine($"No version or invalid version '{project.Version}' found, using default.");
                     version = new SemVersion(0, 1, 0);
                     project.Version = version.ToString();
                 }
 
                 step.MarkComplete();
             }
+
+            return Task.CompletedTask;
+        }
+
+        private static void EnsureMSBuildRegistered(FileInfo projectFile)
+        {
+            if (!registered)
+            {
+                // It says VisualStudio - but we'll just use .NET SDK
+                var instances = MSBuildLocator.QueryVisualStudioInstances(new VisualStudioInstanceQueryOptions()
+                {
+                    DiscoveryTypes = DiscoveryType.DotNetSdk,
+
+                    // Using the project as the working directory. We're making the assumption that
+                    // all of the projects want to use the same SDK version. This library is going
+                    // load a single version of the SDK's assemblies into our process, so we can't
+                    // use supprt SDKs at once without getting really tricky.
+                    //
+                    // The .NET SDK-based discovery uses `dotnet --info` and returns the SDK
+                    // in use for the directory.
+                    //
+                    // https://github.com/microsoft/MSBuildLocator/blob/master/src/MSBuildLocator/MSBuildLocator.cs#L320
+                    WorkingDirectory = projectFile.DirectoryName,
+                });
+
+                var instance = instances.SingleOrDefault();
```

Comment:
Will this ever return more than one? Should we do `FirstOrDefault` instead?

Reply:
It will always return one. It does `dotnet --info` and then returns the SDK would be used if you ran `dotnet`. If it starts to return multiple, we want to know because that would cause bugs - we'd need to know how to choose between different versions.